packages/core/src/utils/formatQuery/formatQuery.ts

```diff
@@ -524,12 +527,85 @@ function formatQuery(
     return ruleGroupProcessor_option(ruleGroup, finalOptions);
   }
 
+  // Helper function to filter out muted items
+  const filterMuted = <T extends RuleGroupTypeAny>(query: T): T => {
```

Comment:
This feels unnecessary. Instead of recursing through the query object one extra time on every pass just to cover a rare use case, I think it would be better to add the logic to the existing rule group processors (as was already done with `defaultRuleGroupProcessorSQL`).

If you did want to process the query object before sending it to `formatQuery`, I would recommend doing it in userland, and I would recommend using [`transformQuery`](https://react-querybuilder.js.org/docs/utils/misc#transformquery).